ios/FluentUI/List/ListItem.swift

```diff
@@ -0,0 +1,209 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
+
+public typealias ListItemAccessoryType = TableViewCellAccessoryType
+public typealias ListItemBackgroundStyleType = TableViewCellBackgroundStyleType
+public typealias ListItemCustomViewSize = MSFTableViewCellCustomViewSize
+public typealias ListItemTokenSet = TableViewCellTokenSet
+
+/// View that represents an item in a List.
+public struct ListItem<LeadingView: View,
+                       TrailingView: View,
+                       Title: StringProtocol,
+                       Subtitle: StringProtocol,
+                       Footer: StringProtocol>: View {
+
+	// MARK: Initializer
+
+	/// Creates a ListItem view
+	/// - Parameters:
+	///   - title: Text that appears as the first line of text
+	///   - subtitle: Text that appears as the second line of text
+	///   - footer: Text that appears as the third line of text
+	///   - leadingView: The view that appears on the leading edge of the view
+	///   - trailingView: The view that appears on the trailing edge of the view, next to the accessory type if provided
+    public init(title: Title,
+                subtitle: Subtitle = String(),
+                footer: Footer = String(),
+                @ViewBuilder leadingView: @escaping () -> LeadingView,
+                @ViewBuilder trailingView: @escaping () -> TrailingView) {
+        self.title = title
+        self.subtitle = subtitle
+        self.footer = footer
+        self.leadingView = leadingView()
```

Comment:
Do we know if there are perf implications of calling this on init, instead of storing the ViewBuilder and only calling it from body? I'd gravitate towards the latter to be safe. Init can get called *much* more than view body.

Reply:
so the trade off was having default values in the init like `leadingView = { EmptyView()}` or not. @ViewBuilder cant have an optional value, so this was trying to address how we could have leadingView and trailingView as nil in the private variable instead of creating `EmptyView()`


This conversation from the first iteration: https://github.com/microsoft/fluentui-apple/pull/1845#discussion_r1274263135

Reply:
ohh I see. I think you could store the private vars as `var content: (() -> Content)?` - that would allow you to set them to nil from those other inits, and still defer the actual calling of the closure to view body. Up to you tho, I'm not even sure about the perf impact.

Reply:
ah gotcha. I think that will work, and makes sense